src/Microsoft.AspNet.Mvc.ModelBinding/Internal/ModelBindingHelper.cs

```diff
@@ -100,7 +100,20 @@ internal static void AddModelErrorBasedOnExceptionType(ModelBindingContext bindi
         {
             if (IsFormatException(ex))
             {
-                bindingContext.ModelState.TryAddModelError(bindingContext.ModelName, ex.Message);
+                var modelState = bindingContext.ModelState[bindingContext.ModelName];
+                if (modelState == null)
+                {
+                    bindingContext.ModelState.TryAddModelError(bindingContext.ModelName, ex.Message);
```

Comment:
This seems wierd/bad. We don't want to blindly add exception messages to modelstate. We don't control most of these messages, they come from the CLR or user code.